runtime/src/main/java/org/corfudb/runtime/object/transactions/WriteAfterWriteTransactionalContext.java

```diff
@@ -45,6 +45,18 @@
      */
     @Override
     public long commitTransaction() throws TransactionAbortedException {
+        long ret = commitTransactionNoReleaseLock();
+
+        // now unlock this transaction's contention lock
+        try {
+            getMTxLock().unlock();
+        } catch (IllegalMonitorStateException me) {}
```

Comment:
Since this is an error (and not a normal condition, like interruptedexception is, I think you need to log an error level message.